Makefile

```diff
@@ -1,37 +1,61 @@
-.ONESHELL:
 .PHONY:
 
 all: install
 
 ##### Variables ######
 COLOR := "\e[1;36m%s\e[0m\n"
 
+# Disable cgo by default.
+CGO_ENABLED ?= 0
+
 TEMPORAL_ROOT := temporal
 TCTL_ROOT := tctl
+CLI_ROOT := cli
+DOCKERIZE_ROOT := dockerize
 IMAGE_TAG ?= sha-$(shell git rev-parse --short HEAD)
 TEMPORAL_SHA := $(shell sh -c 'git submodule status -- temporal | cut -c2-40')
 TCTL_SHA := $(shell sh -c "git submodule status -- tctl | cut -c2-40")
 
 DOCKER ?= docker buildx
 BAKE := IMAGE_TAG=$(IMAGE_TAG) TEMPORAL_SHA=$(TEMPORAL_SHA) TCTL_SHA=$(TCTL_SHA) $(DOCKER) bake
+NATIVE_ARCH := $(shell go env GOARCH)
 
 ##### Scripts ######
 install: install-submodules
+clean:
+	rm -rf ./build
 
 update: update-submodules
 
 install-submodules:
-	@printf $(COLOR) "Installing temporal and tctl submodules..."
-	git submodule update --init $(TEMPORAL_ROOT) $(TCTL_ROOT)
+	@printf $(COLOR) "Installing submodules..."
+	git submodule update --init
 
 update-submodules:
 	@printf $(COLOR) "Updatinging temporal and tctl submodules..."
 	git submodule update --force --remote $(TEMPORAL_ROOT) $(TCTL_ROOT)
 
 ##### Docker #####
 
-build:
-	$(BAKE)
+# If you're new to Make, this is a pattern rule: https://www.gnu.org/software/make/manual/html_node/Pattern-Rules.html#Pattern-Rules
+# $* expands to the stem that matches the %, so when the target is amd64-bins $* expands to amd64
+%-bins:
+	mkdir -p build/$*
+	cd $(DOCKERIZE_ROOT) && CGO_ENABLED=$(CGO_ENABLED) GOOS=linux GOARCH=$* go build -o dockerize .
+	@cp $(DOCKERIZE_ROOT)/dockerize build/$*/
+	@GOOS=linux GOARCH=$* CGO_ENABLED=$(CGO_ENABLED) make -C $(TEMPORAL_ROOT) bins
+	@cp $(TEMPORAL_ROOT)/temporal-server build/$*/
+	@cp $(TEMPORAL_ROOT)/temporal-cassandra-tool build/$*/
+	@cp $(TEMPORAL_ROOT)/temporal-sql-tool build/$*/
+	@cp $(TEMPORAL_ROOT)/tdbg build/$*/
+	@GOOS=linux GOARCH=$* CGO_ENABLED=$(CGO_ENABLED) make -C $(CLI_ROOT) build
+	@cp ./$(CLI_ROOT)/temporal build/$*/
+	@GOOS=linux GOARCH=$* CGO_ENABLED=$(CGO_ENABLED) make -C $(TCTL_ROOT) build
+	@cp ./$(TCTL_ROOT)/tctl build/$*/
+	@cp ./$(TCTL_ROOT)/tctl-authorization-plugin build/$*/
```

Comment:
same for both of these. actually the new cli doesn't even have a makefile, so you'll have to use go install there